### PR TITLE
GH-295 Fix priority_resolution attribute for npm, cargo, docker, helm…

### PR DESCRIPTION
**Pull request by maheshjfrog**
… remote repositories
Fixes: #295

### CODE REVIEW

**Review by alexhung — CHANGES_REQUESTED** [review]
Code looks good. Missing CHANGELOG.md though 😄

**Comment by maheshjfrog** [comment]
> 

Will add the CHANGELOG.md

File: pkg/artifactory/resource_artifactory_remote_pypi_repository.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+		Optional:    true,
+		Default:     "simple",
+		Description: "Usually should be left as a default for 'simple', unless the remote is a PyPI server that has custom registry suffix, like +simple in DevPI",
+	},
+})
+
+type PypiRemoteRepo struct {
+	RemoteRepositoryBaseParams
+	RegistryUrl      string `hcl:"pypi_registry_url" json:"pyPIRegistryUrl"`
+	RepositorySuffix string `hcl:"pypi_repository_suffix" json:"pyPIRepositorySuffix"`
```
Comment (chb0github) [inline pkg/artifactory/resource_artifactory_remote_pypi_repository.go:27]: Is there a reason no one allows the default packer to general the hcl?

File: pkg/artifactory/resource_artifactory_remote_pypi_repository.go [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+var pypiRemoteSchema = mergeSchema(baseRemoteSchema, map[string]*schema.Schema{
+	"pypi_registry_url": {
+		Type:         schema.TypeString,
+		Optional:     true,
+		Default:      "https://pypi.org",
+		ValidateFunc: validation.IsURLWithHTTPorHTTPS,
```
Comment (chb0github) [inline pkg/artifactory/resource_artifactory_remote_pypi_repository.go:13]: Please use the new `ValidateDiagFunc`

**Review by chb0github — CHANGES_REQUESTED** [review]
see comments

**Review by alexhung — APPROVED** [review]

**Pull request merged**